src/coord/src/coord.rs

```diff
@@ -772,6 +772,16 @@ impl Coordinator {
         self.broadcast(dataflow::Command::AdvanceSourceTimestamp { id, update });
     }
 
+    fn message_scrape_metrics(&mut self) {
+        let metric_scraper = self
+            .metric_scraper
+            .as_mut()
+            .expect("received Message::ScrapeMessage without metric scraper");
```

Comment:
I feel this could log an `error` and continue doing nothing? It would indicate a bug, but we probably shouldn't panic the system if we run into this condition.

Reply:
Ack, yeah, agree it's not great to panic there. I refactored so that there is always a metric scraper, but we just don't produce any `Message::ScrapeMessage`s if `introspection-granularity=off`.

Reply:
That is even better! Nice.